tripy/nvtripy/frontend/ops/shape.py

```diff
@@ -49,7 +49,8 @@ def shape(self: "nvtripy.Tensor") -> ShapeLike:
     if all(dim >= 0 for dim in self.trace_tensor.shape) and not self.trace_tensor.is_compile_tracer:
         return self.trace_tensor.shape
 
+    shape = op_utils.create_op(Shape, [self])
     return [
-        op_utils.create_op(GetDimensionSize, [self], dim=index, always_cast_to_dimension_size=True)
+        op_utils.create_op(GetDimensionSize, [shape], dim=index, always_cast_to_dimension_size=True)
```

Comment:
Wondering if I should change the name of `GetDimensionSize` - it's now basically just a special kind of slice op. Maybe `ExtractDimension` instead?